svc.go

```diff
@@ -96,6 +103,21 @@ func (as *AutoNATService) handleStream(s network.Stream) {
 	}
 }
 
+// Optimistically extract the net.IP host from a multiaddress.
+func addrToIP(addr ma.Multiaddr) net.IP {
+	if v4, err := addr.ValueForProtocol(ma.P_IP4); err == nil {
+		if c, err := ma.NewComponent(ma.ProtocolWithCode(ma.P_IP4).Name, v4); err == nil {
+			return net.IP(c.RawValue())
```

Comment:
As we discussed this is probably fine since we check if it's a circuit address first and AFAIK that's the only multiaddr protocol we have that would result in multiple IPv4 or IPv6 codes being in a single multiaddr.

Reply:
Switched this logic to more reasonably use the existing logic in `manet` to initially learn the host.